devices/motorola-potter/default.nix

```diff
@@ -0,0 +1,71 @@
+{ config, lib, pkgs, ... }:
+
+{
+  mobile.device.name = "motorola-potter";
+  mobile.device.identity = {
+    name = "Moto G5 Plus";
+    manufacturer = "Motorola";
+  };
+
+  mobile.hardware = {
+    soc = "qualcomm-msm8953";
+    ram = 1024 * 3;
+    screen = {
+      width = 1080; height = 1920;
+    };
+  };
+
+  mobile.boot.stage-1 = {
+    kernel.package = pkgs.callPackage ./kernel { };
+  };
+
+  mobile.device.firmware = pkgs.callPackage ./firmware {};
+
+  hardware.firmware = [
+    (config.mobile.device.firmware.override {
+      modem = ../../vendor/modem;
+    })
+  ];
+
+  mobile.system.android.device_name = "potter";
+  mobile.system.android.bootimg = {
+    flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  # Using `xz` allows us to fit the kernel + dt + initrd in 16MiB
+  #   Kernel: ~8.8M
+  #   DT:      1.2M
+  # We're left with ~6MB for the compressed initrd.
+  mobile.boot.stage-1.compression = lib.mkDefault "xz";
```

Comment:
Since you were talking about the boot.img being too big, I assume this is not the numbers for your system.